### DISCUSSION TITLE
potential bug in ComputeStVenantKirchhoffStress & NeoHookeanStress?

### DISCUSSION THREAD

**Discussion by rcontinuum** [idaholab/moose#22581] [Q&A Modules: Solid mechanics] [answered]
Hello, 

in the new Lagrangian Kernel/Materials there may be a bug for the case `large_kinematics = false`. E.g., look at 

```
void
ComputeStVenantKirchhoffStress::computeQpPK2Stress()
{
  ...
  if (_large_kinematics) // Large deformations = Green-Lagrange strain
    strain = _E[_qp];
  else // Small deformations = linear strain
    strain = 0.5 * (_F[_qp] + _F[_qp].transpose());      // <=====???=========================  

  // The stress update is linear with the correct strains/frame
  _S[_qp] = _C[_qp] * strain;
}
```
To my understanding the marked linearized strain computation should be replaced by something like  

```
const RankTwoTensor I = RankTwoTensor::Identity();
RankTwoTensor H = _F[_qp] - I;   // displacement gradient
RankTwoTensor strain  = 0.5 * (H + H.transpose()); 
```
or put it in one line.
You can check that by setting `large_kinematics = false` (also in the [Materials] section)  in, e.g., the file 
`moose/modules/tensor_mechanics/test/tests/lagrangian//materials/correctness/stvenantkirchhoff.i`
and set `exodus = true`. Then one observe in paraview a much too large deformation in comparison to the case `large_kinematics = true`. 

The same problem applies to `void ComputeNeoHookeanStress::computeQpPK2Stress()`

Cheers

**Comment by GiudGiud** [comment]
@reverendbedford

**Comment by reverendbedford** [comment, marked as answer]
Yes, this is a bug.  I opened issue #22588 and will fix it.

**Reply by rcontinuum** [reply]
Thank you!
